pkg/features/features.go

```diff
@@ -111,6 +112,9 @@ const (
 	MutableShootSpecNetworkingNodes featuregate.Feature = "MutableShootSpecNetworkingNodes"
 )
 
+// DefaultFeatureGate TODO: Comment
+var DefaultFeatureGate = utilfeature.DefaultMutableFeatureGate
```

Comment:
I would suggest to drop this commit, please. So far, we preferred individual feature gates per component/binary, and there should be no good reason to change this now. At least not as part of this PR. If you still think it's valuable, please trigger a separate/dedicated discussion about it via opening an according issue.

Reply:
This PR does not work if this commit is not included here unfortunately...
The problem: while validating the config.yaml for the gardenlet, we use the Seed validation since there is a seed object in the config that configures the gardenlet. In this config there is also the definition on which features gates should be activated and which not.
The validation of the Seed object requires the SingleStack feature gate to be activated but we haven't yet activated the gate since we are still validating the config.yaml, thus we haven't applied it.
Would you suggest creating another PR on which this one here depends?

Reply:
Hold your horses for now - as said above, I don't even know whether we want this. I believe it requires a discussion first.
Where is this validation code that you mentioned which requires all this? I tried to find it in the PR/commits, but wasn't successful. Maybe we can find a solution without it or at least a compromise to not block this PR.

Reply:
Okay i will open a discussion for this :)
The respective code is here https://github.com/gardener/gardener/blob/931b37c85f6ea6ec5c5e898a3e9c4235b628db7d/pkg/apis/core/validation/utils.go#L192 and it got introduced in an earlier PR where all IPv6 types got introduced (https://github.com/gardener/gardener/pull/7288/files).
The validation code was meant to be used in the apiserver to validate seed objects, but as stated, the gardenlet got a seed object inside its startup configuration.

Reply:
Would it be an option to 
- move this function to https://github.com/gardener/gardener/tree/master/pkg/utils/validation/cidr (possibly the `cidr` package could be renamed to `networking` or similar)
- pass a `bool` to the function describing whether IPv6 single stack should be allowed?

Then the respective components could pass their feature gates accordingly to the `bool` argument?

Reply:
There were some misunderstandings here: with the change we use a central feature gate map shared across all component, but the components still register their individual feature gates.
This doesn't change that feature gates are only available on the components for which they are relevant.
However, we can now share code that checks a certain feature gate's state across components, which wasn't possible before.

I explained this in a code comment in a new commit :)